core/client/gobindClient/client.go

```diff
@@ -0,0 +1,174 @@
+// TODO(amarcedone) : rename package to gobind_client after https://github.com/golang/go/issues/17359 is solved.
+package gobindClient
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/keytransparency/cmd/keytransparency-client/grpcc"
+	"github.com/google/trillian"
+	"github.com/google/trillian/client"
+	"github.com/google/trillian/crypto/keys/pem"
+	"github.com/google/trillian/merkle/hashers"
+	_ "github.com/google/trillian/merkle/objhasher" // Used to init the package so that the hasher gets registered (needed by the bGetVerifierFromParams function)
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"net"
+	"time"
+
+
+	"crypto/tls"
+	"crypto/x509"
+	"github.com/google/keytransparency/core/client/kt"
+	"github.com/google/keytransparency/core/crypto/keymaster"
+	"github.com/google/keytransparency/core/crypto/vrf/p256"
+	pb "github.com/google/keytransparency/impl/proto/keytransparency_v1_service"
+	"log"
+)
+
+type BClientParams struct {
+	KtURL        string
+	MapID        int64
+	KtTLSCertPEM []byte
+	VrfPubPEM    []byte
+	KtSigPubKey  []byte
+	LogPEM       []byte
+}
+
+// TODO(amarcedone) consider persisting the client or at least the trusted smr, to gain efficiency and stronger consistency guarantees.
+
+func NewBClientParams(KtURL string, MapID int64, KtTLSCertPEM, VrfPubPEM, KtSigPubKey, LogPEM []byte) *BClientParams {
+	// Note: byte arrays need to be explicitly cloned due to some gobind limitations.
+	cKtTLSCertPEM := make([]byte, len(KtTLSCertPEM))
+	copy(cKtTLSCertPEM, KtTLSCertPEM)
+	cVrfPubPEM := make([]byte, len(VrfPubPEM))
+	copy(cVrfPubPEM, VrfPubPEM)
+	cKtSigPubKey := make([]byte, len(KtSigPubKey))
+	copy(cKtSigPubKey, KtSigPubKey)
+	cLogPEM := make([]byte, len(LogPEM))
+	copy(cLogPEM, LogPEM)
+
+	return &BClientParams{KtURL, MapID, cKtTLSCertPEM, cVrfPubPEM, cKtSigPubKey, cLogPEM}
+}
+
+func BGetEntry(timeoutInMilliseconds int, clientParams *BClientParams, userID, appID string) ([]byte, error) {
```

Comment:
It can be a proper class (i.e. one can use the syntax BClientParams.MyMethod(myparam) ), but to be directly visible from the Java code it has to only contain members which are gobind friendly. 

Another option to increase performance would be to create a client and save it as a global variable of this package, and have GetEntry just use this global client to make requests.
On the java side, we can ensure that there is only one copy of the client available (so only one set of client parameters such as ktaddress/verification keys used can be used at the same time).
Even more, we could use a global variable to store a map of clients (essentially getting around the gobind limitation of not being able to pass around references to arbitrary objects by creating our own reference mechanism), but not sure if that is worth it. Probably one variable as reference would be good for now.

Any thoughts?